### ISSUE TITLE
Checkbox: onClick doesn't fire when inside a Dropdown

### ISSUE THREAD

**Issue by mihai-dinculescu** [Semantic-Org/Semantic-UI-React#2121]
### Steps
Add a checkbox inside a dropdown, either as header or item.

### Expected Result
- `onClick` fires

### Actual Result
- `onClick` doesn't fire
- also note the weird font

### Version
0.74.1

### Testcase
https://codepen.io/anon/pen/zENBPx

**Comment by levithomason** [comment]
Looks like the Dropdown is overriding `onClick` when looping over the `options` and creating `DropdownItem`s.  Seen [here](https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/modules/Dropdown/Dropdown.js#L1213).

### Workaround

Use `onMouseDown` instead.  Working fork https://codepen.io/levithomason/pen/qPRqRg.

### Fix

The `DropdownItem.create()` factory call needs to no override the `onClick` handler, but wrap it.  An example of this is shown in the same file, **Dropdown.js**, where `handleIconOverrides()` is used to wrap the icon handlers and not override them.

**Comment by levithomason** [comment]
Note, children are not supported in a Dropdown.  Use the `options` prop instead.  The children API will be removed.

**Comment by mihai-dinculescu** [comment]
I'll have a go at getting it fixed.

**Comment by mihai-dinculescu** [comment]
@levithomason I don't think that DropdownItem is the issue. The Checkbox is displayed inside a DropdownHeader after all, not inside an Item. `handleItemClick` is not called.

Adding `multiple={true}` fixes the problem, but it's something I'd prefer not to do.

**Comment by ghost** [comment]
don't close this issue `bot`

**Comment by tarang9211** [comment]
@levithomason or anyone here, what's the update on this issue? any ideas on how to fix?

**Comment by nwpray** [comment]
The this.toggle(e) @ src/modules/Dropdown/Dropdown.js:705 appears to be whats actually closing the dropdown on click. I'll look into a fix

![image](https://user-images.githubusercontent.com/11433893/38714903-a471473a-3e95-11e8-8c91-6605574576eb.png)

**Comment by nwpray** [comment]
I noticed that when you click on other Items or Headers, the order of called Dropdown methods are:

![image](https://user-images.githubusercontent.com/11433893/38717821-80e110e6-3ea6-11e8-95f2-80e0417dc0b0.png)

But when you click on the checkbox its order is:

![image](https://user-images.githubusercontent.com/11433893/38717842-a04a35c0-3ea6-11e8-9f63-2c00bef99771.png)

So it appears that the `blur` event is being called before the `mouseDown` event which causes the `close` method of Dropdown to be called without the `isMouseDown` property being set in the `handleMouseDown` method. I tried with vanilla html and the order is mouseDown->blur so I am a bit confused. I will keep looking into this.

**Comment by walter76** [comment]
Hi,

I gave this issue a shot and implemented what was suggested by @levithomason in a previous comment. But this did not work as the onClick handler will never be called. I checked this by adding a debug() output into onClick.

As it was also suggested to not use children, but instead use the options property, I changed the example to do so. But this also did not change anything. The onClick handler on the DropdownItem is simply not called. My guess is, that handleMouseDown() captures the event so it is not being further propagated.

I could hook into handleMouseDown() and propagate onClick, but this kind of feels like a hack. Are there any other ideas?

**Comment by ghost** [comment]
I tried to replicate the workaround @levithomason shared and it worked for me. 

[Sandbox](https://codesandbox.io/s/y73wwn4l4j)

**Comment by Shenrak** [comment]
I would like to trigger a simple onClick : 

`public onClick = () => this.setState({loading: true})`

That would display the loading prop of the dropdown object onOpen, but it doesn't seem to work.
onOpen, onMouseDown, onFocus don't work either...

Edit : onOpen is perfectly working, i was too dumb to check my shouldComponentUpdate. Sorry.

**Comment by levithomason** [comment]
Note, I've given a workaround and a proper fix here https://github.com/Semantic-Org/Semantic-UI-React/issues/2121#issuecomment-331966524.  I've also included a reference to the proper pattern in another component that can be copied.  Any PR implementing this would be much appreciated!
